File: code/backend/core/src/main/java/games/paths/core/port/in/EchoPort.java
package games.paths.core.port.in;

import java.util.Map;

/**
 * EchoPort - Inbound port for server health/status checks.
 * Part of the Hexagonal Architecture: this is a domain port
 * that the adapter-rest module will call.
 */
public interface EchoPort {
    String getServerStatus();
    long getTimestamp();
    Map<String, String> getServerProperties();
}

File: code/backend/python/app/core/ports/echo_port.py
from abc import ABC, abstractmethod

class EchoPort(ABC):
    @abstractmethod
    def get_server_status(self) -> str:
        """Returns the current server status."""
        pass

    @abstractmethod
    def get_timestamp(self) -> int:
        """Returns the current server timestamp in milliseconds."""
        pass

    @abstractmethod
    def get_server_properties(self) -> dict[str, str]:
        """Returns the server properties."""
        pass
